File: growing-shop-auth/src/main/java/com/example/growingshopauth/config/security/UserAuthentication.java
package com.example.growingshop.global.config.security;

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collections;

public class UserAuthentication extends UsernamePasswordAuthenticationToken {
    public UserAuthentication(Object principal, GrantedAuthority authority) {
        super(principal, null, Collections.singletonList(authority));
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/config/security/UserAuthentication.kt
package com.example.growingshopauth.config.security

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

class UserAuthentication(principal: Any, authority: GrantedAuthority) :
    UsernamePasswordAuthenticationToken(principal, null, listOf(authority))

